scripts/validate-charts

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+set -e
+
+info() {
+    echo '[INFO] ' "$@"
+}
+
+error() {
+    echo '[ERROR] ' "$@" >&2
+}
+
+fatal() {
+    echo '[ERROR] ' "$@" >&2
+    exit 1
+}
+
+cleanup() {
+  exit_code=$?
+  trap - EXIT INT
+  rm -rf  /tmp/tmp.*.tar.gz
+  exit ${exit_code}
+}
+trap cleanup EXIT INT
+
+
+download_chart() {
+    chart_version=$1
+    chart_name=$2
+    bootstrap=$3
+
+    chart_package=${chart_name%%-crd}
+
+    chart_url=${CHART_REPO:="https://rke2-charts.rancher.io"}/assets/${chart_package}/${chart_name}-${chart_version:="v0.0.0"}.tgz
+
+    chart_tmp=$(mktemp --suffix .tar.gz)
+    
+    curl -fsSL "${chart_url}" -o "${chart_tmp}"
+
+    echo $chart_tmp
+}
+
+check_system_registry() {
+    chart_version=$1
+    chart_name=$2
+    chart_tmp=$3
+
+    yaml_tmp=$(mktemp --suffix .yaml)
+
+    values="global.systemDefaultRegistry=my-registry,global.cattle.systemDefaultRegistry=my-registry"
+    if [[ $chart_name == 'rancher-vsphere-csi' ]]; then
+        values="$values,vCenter.clusterId=test-id"
+    fi
+    helm template test-chart --set $values $chart_tmp > $yaml_tmp;
+
+    awk '$1 ~ /^image:/ {
+        if( $2 !~ /my-registry/ && $2 !~ busybox) {
+            print $2
+        }
+    }
+    ' $yaml_tmp
+}
+
+check_airgap() {
+    chart_version=$1
+    chart_name=$2
+    chart_tmp=$3
+
+    if [[ $chart_name == 'rancher-vsphere-cpi' ]]; then
```

Comment:
is this chart untestable? if so, why?

Reply:
the chart doesn't work well with `helm template` because of the `versionOverrides` in its `values.yaml`.
Since there is no actual k8s cluster to determine the version of the image to use, `helm template` uses the `.Capabilities.APIVersion` of the version of k8s that helm was built against so it chooses an image that doesn't match the one we need in the airgap file.

To workaround this we would need manually install the right version of Helm which would be different for the main branch and each release branch and would need to be updated for every new k8s version. Since it's only for this chart, I found it simpler to skip the test.

See https://github.com/rancher/vsphere-charts/blob/7c1f2d4ee52548bcc8f7aee0e986b338a229ac8a/charts/rancher-vsphere-cpi/values.yaml#L28

Reply:
You don't need to install different versions of Helm, you just need to use the `--kube-version` flag to specify the Kubernetes version you want it to use when templating. You could probably just pass that the `$VERSION` environment variable to make sure the chart handles it properly.